ibis/expr/api.py

```diff
@@ -1031,4 +1032,40 @@ def get_backend(expr: Expr | None = None) -> BaseBackend:
 intersect = ir.Table.intersect
 difference = ir.Table.difference
 
-_ = Deferred()
+_ = deferred = Deferred()
+"""Deferred expression object.
+
+Use this object to refer to a previous table expression in chain of
+expressions.
+
+!!! note "`_` may conflict with other idioms in Python"
+
+    See https://github.com/ibis-project/ibis/issues/4704 for details.
+
+    Use `from ibis import deferred as <NAME>` to assign a different name to
+    the deferred object builder.
+
+Examples
+--------
+>>> from ibis import _
+>>> t = ibis.table(dict(key="int", value="float"), name="t")
+>>> t.group_by(key=_.key - 1).agg(total=_.value.sum())
```

Comment:
On the other hand, recommending they use `_` everywhere would avoid the "why does `t` work here but not later on" issue. No strong thoughts.

Reply:
Well, I just accidentally clobbered @gforsyth's commit. I'm going to leave it as is, because @jcrist makes a good point and I think we should avoid giving the impression that `t` should work everywhere, `deferred`-first IOW.